### DISCUSSION TITLE
Running clean architecture with IIS

### DISCUSSION THREAD

**Discussion by ttt15** [jasontaylordev/CleanArchitecture#488] [Q&A]
Hello,
I want my project that is based on CleanArchitecture to run with IIS. In the CleanArchitecture project I declared WebUI as start project and clicked 'Publish'. Target location is c:\web\clean. This works. In the IIS manager I specified the physical path(c:\web\clean) and as binding(http://192.168.178.10:8080). Then I started the website. However, when I go to the website I get an error message: "HTTP Error 500.30 - ASP.NET Core app failed to start ".
I have run a simple ASP.NET Core web app with IIS for testing. It works. Used the same steps as CleanArchitecture.
Is there a tutorial somewhere on how to run projects like the CleanArchitecture project with IIS? Don't know much about IIS and web servers.

Best regards

**Comment by ghost** [comment]
Hello @ttt15 ,

Quick suggestion to try. In `launchSettings.json` make sure the port specified in the application URL matches your IIS setup. So for example in `main` the URL is specified as `http://localhost:5000`. In your project, change this to `http://localhost:8080` so it matches your IIS setup.  Something like this:

```json
"CleanArchitecture.WebUI": {
      "commandName": "Project",
      "launchBrowser": true,
      "environmentVariables": {
        "DOTNET_ENVIRONMENT": "Development"
      },
      "applicationUrl": "https://localhost:5001;http://localhost:8080"
    },
```

Snippet above was pulled [from main](https://github.com/jasontaylordev/CleanArchitecture/blob/main/src/WebUI/Properties/launchSettings.json).

**Comment by ttt15** [comment]
```
```
Hello @davidmcenedella,
unfortunatly it doesn't work. First I got error message regarding to a lock in system.webServers/handlers(http error 500.19). After unlocking the handlers with "%windir%\system32\inetsrv\appcmd.exe unlock config -section:system.webServer/handlers -commitpath:apphost", I get 
```
HTTP Error 500.0 - ASP.NET Core IIS hosting failure (in-process)
Troubleshooting steps:

    Check the system event log for error messages
    Enable logging the application process' stdout messages
    Attach a debugger to the application process and inspect

For more information visit: https://go.microsoft.com/fwlink/?LinkID=2028526"
```
Nothing in the logs. Maybe I have to attach/start VS as debugger in the code. But I have the feeling that the debugger won't help me.

**Comment by ttt15** [comment]
A network-related or instance-specific error occurred while establishing a connection to SQL Server...
Piped logs to a file. Will work on this problem.

**Comment by ttt15** [comment]
Ok. Now it's running.
This my POC.
I want a publish the clean architecture stuff as https://abc.com:5001 with a self signed certificate.
0. Configured DNS
1. Created a ssl certificate:
```
openssl req -x509 -newkey rsa:4096 -sha256 -nodes -keyout abc.key -out abc.crt -subj "/CN=abc.com" -days 3650
openssl pkcs12 -export -out abc.pfx -inkey abc.key -in abc.crt -certfile abc.crt
```

2. Imported the certifcate: 
https://www.youtube.com/watch?v=HHU46l6G2oc

3. Modified some config file the CleanArchitecture project:
\src\WebUI\appsettings.json: 
```
"UseInMemoryDatabase": true,
```

\src\WebUI\Properties\launchSettings.json:
```
"template.WebUI": {
      "commandName": "Project",
      "launchBrowser": true,
      "environmentVariables": {
        "DOTNET_ENVIRONMENT": "Production"
      },
      "applicationUrl": "https://abc.com:5001"
    },
```
\src\WebUI\appsettings.Production.json:
```
"IdentityServer": {
    "Key": {
      "Type": "Store",
      "StoreName": "My",
      "StoreLocation": "LocalMachine",
      "Name": "CN=abc.com"
    }
  }
```
4. Create the project with dotnet publish:
```
publish -o c:\publish -r win-x64 --nologo --self-contained true -c Release
```
5. Created a new website in IIS:
Sitename: test
Path: c:\publish
Type: https
IP: IP of the server
Port: 5001
Hostname: abc.com
SSL-Certificate: abc.com

**Reply by ghost** [reply]
Good stuff. Glad to hear you got it going.

**Reply by ttt15** [reply]
Thanks!

**Reply by ITEAlaaEddin** [reply]
What should I change in the Angular so that it uses the same certificate key?
